R/utilities-data-combined.R

```diff
@@ -240,7 +243,7 @@ calculateResiduals <- function(dataCombined,
   #
   # To avoid this, just remove rows where any of the quantities are `0`s.
   if (scaling %in% c(tlf::Scaling$log, tlf::Scaling$ln)) {
-    pairedData <- dplyr::filter(pairedData, xValues != 0, yValues != 0)
+    pairedData <- dplyr::filter(pairedData, xValues != 0, yValues != 0, predictedValues != 0)
```

Comment:
Something else comes in my mind now where I see this:

in PK-Sim/MoBi, log residuals are calculated via the defined there `LogSafe `function:
https://github.com/Open-Systems-Pharmacology/OSPSuite.Core/blob/fcd42c611764e4af44f266285175ef66f706fdde/src/OSPSuite.Core/Domain/Services/ParameterIdentifications/ResidualCalculator.cs#L118-L124

with
https://github.com/Open-Systems-Pharmacology/OSPSuite.Core/blob/fcd42c611764e4af44f266285175ef66f706fdde/src/OSPSuite.Core/Domain/Constants.cs#L92

The log residual itself is then calculated as 
https://github.com/Open-Systems-Pharmacology/OSPSuite.Core/blob/fcd42c611764e4af44f266285175ef66f706fdde/src/OSPSuite.Core/Domain/Services/ParameterIdentifications/ResidualCalculatorForOnlyObservedData.cs#L12-L20

This soves different problems, e.g. if simulated values are slightly negative (numerical zero) log would return NaN; if simulated values are positive but extremely small (numerical zero as well), residual might become infinity (or minus infinity) etc.

So how about implementing residual calculation in R in the same way (just with `ln` instead of `log10`)?

Reply:
Why aren't we implementing the same function in R and ospsuite.core? This does not make sense to me to have two different implementations for the same concept. If the one we have is not wrong, we should stick with this one for R also. I remember that we had long discussions about this in the past (also with @ju-rgen ) and the log10 was the defacto choice

Reply:
well, seems there are different opinions on this: https://github.com/Open-Systems-Pharmacology/OSPSuite-R/issues/1087

Reply:
I saw and the point is that the other implementation is also correct. (@sfrechen says so himself)
I think having two different implementations is simply wrong. We should keep with one to show that we can at least be cohesive within our suite. If that means changing the one in PK-Sim. So be it. But maybe @ju-rgen remembers why this was chosen back then.

Reply:
This was also checked with Roland at the time. The documentation also stipulates log10.... just as an FYI

Reply:
Okay, I think then we should also switch to using `log10()` in both ospsuite and reporting engine packages.

I will update the PR for the same.

Reply:
@IndrajeetPatil Please don't switch to `log10 `until we have the final decision on this.
It was `ln `in RE all the time (also in the previous MAtlab version) and `log10 `all the time in PK-Sim.
I agree that it would be good to have the same algo in all software pieces, but for now I could not give the final answer which of both we should use.

Reply:
Okay, in that case, I would appreciate if we don't block the current PR, which is unrelated to the log issue, but rather related to when some predicted values shouldn't be computed.

Reply:
Agree. I created an issue which can be adressed later: https://github.com/Open-Systems-Pharmacology/OSPSuite-R/issues/1091